src/ripple/ledger/impl/Directory.cpp

```diff
@@ -0,0 +1,226 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/ledger/Directory.h>
+
+namespace ripple {
+
+using iterator = Dir::iterator;
+
+Dir::Dir(ReadView const& view,
+        Keylet const& key)
+    : view_(&view)
+    , root_(key)
+    , sle_(view_->read(root_))
+
+{
+    if (sle_ == nullptr)
+        throw std::runtime_error("Invalid AccountID");
+    indexes_ = &sle_->getFieldV256(sfIndexes);
+}
+
+#if defined(_MSC_VER) && _MSC_VER <= 1800
+Dir::Dir(Dir&& other)
+    : view_(std::move(other.view_))
+    , root_(std::move(other.root_))
+    , sle_(std::move(other.sle_))
+{
+}
+
+Dir&
+Dir::operator=(Dir&& other)
+{
+    if (this != &other)
+    {
+        view_ = std::move(other.view_);
+        root_ = std::move(other.root_);
+        sle_ = std::move(other.sle_);
+    }
+    return *this;
+}
+
+#endif
+
+auto
+Dir::begin() const ->
+    iterator
+{
+    auto it = iterator(*view_, root_, root_);
+    it.sle_ = sle_;
+    if (! indexes_->empty())
+    {
+        it.indexes_ = indexes_;
+        it.it_ = std::begin(*indexes_);
+        it.index_ = *it.it_;
```

Comment:
That will just require that another iterator constructor be made for `end()` which doesn't want this part.  Not a big deal either way since this logic is only in one place no matter what.